examples/2d/pixel_grid_snap.rs

```diff
@@ -0,0 +1,173 @@
+            transform: Transform::from_xyz(-40., -20., 2.),
+            ..default()
+        },
+        Rotate,
+        HIGH_RES_LAYERS,
+    ));
+}
+
+fn setup_mesh(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    commands.spawn((
+        MaterialMesh2dBundle {
+            mesh: meshes.add(Mesh::from(shape::Quad::default())).into(),
+            transform: Transform::from_xyz(40., 0., 2.).with_scale(Vec3::splat(32.)),
+            material: materials.add(ColorMaterial::from(Color::BLACK)),
+            ..default()
+        },
+        Rotate,
+        PIXEL_PERFECT_LAYERS,
+    ));
+}
+
+fn setup_camera(mut commands: Commands, mut images: ResMut<Assets<Image>>) {
+    let canvas_size = Extent3d {
+        width: RES_WIDTH,
+        height: RES_HEIGHT,
+        ..default()
+    };
+
+    // this Image serves as a canvas representing the low-resolution game screen
+    let mut canvas = Image {
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size: canvas_size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+            view_formats: &[],
+        },
+        ..default()
+    };
+
+    // fill image.data with zeroes
+    canvas.resize(canvas_size);
+
+    let image_handle = images.add(canvas);
+
+    // this camera renders whatever is on [`PIXEL_PERFECT_LAYER`] to the canvas
+    commands.spawn((
+        Camera2dBundle {
+            camera: Camera {
+                // render before the "main pass" camera
+                order: -1,
+                target: RenderTarget::Image(image_handle.clone()),
+                ..default()
+            },
+            ..default()
+        },
+        InGameCamera,
+        PIXEL_PERFECT_LAYERS,
+    ));
+
+    // spawn the canvas
+    commands.spawn((
+        SpriteBundle {
+            texture: image_handle,
+            ..default()
+        },
+        Canvas,
+        HIGH_RES_LAYERS,
+    ));
+
+    // the "outer" camera renders whatever is on [`HIGH_RES_LAYER`] to the screen.
+    // here, the canvas and one of the sample sprites will be rendered by this camera
+    commands.spawn((Camera2dBundle::default(), OuterCamera, HIGH_RES_LAYERS));
+}
+
+/// Rotates entities to demonstrate grid snapping.
+fn rotate(time: Res<Time>, mut query: Query<&mut Transform, With<Rotate>>) {
+    for mut transform in &mut query {
+        let dt = time.delta_seconds();
+        transform.rotate_z(dt);
+    }
+}
+
+/// Scales camera projection to fit the window (integer multiples only).
+fn fit_canvas(
+    mut resize_events: EventReader<WindowResized>,
+    mut q: Query<&mut OrthographicProjection, With<OuterCamera>>,
```

Comment:
nit: our other examples generally use `query` or e.g. `projections` for names like this.